snappy/app/netfilter.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snappy
+
+import (
+	"log"
+	"net"
+	"net/http"
+)
+
+// NetFilter manages an IP-based filter to limit access to Snapweb
+type NetFilter struct {
```

Comment:
could be interesting to add a lock to the struct, you are assuming that it will be e.g. set and never touched again at "runtime"